### PR TITLE
Introduce TOTP mode

### PR DESCRIPTION
**Pull request by vaygr**
This feature is stolen from [passhole](https://github.com/Evidlo/passhole).

Sometimes you might want to use keepmenu as a simple 2FA manager like [2fa](https://github.com/rsc/2fa), while *still using* the same database saving you a few extra clicks.

### CODE REVIEW

**Comment by firecat53** [comment]
Nice. Appreciate all your input recently! Just FYI, I'm going to be a little bit slow and getting through your reports and PRs. Not a lot of time at the moment.

**Comment by vaygr** [comment]
No worries at all! I'll keep myself busy with suggesting more and maybe implementing some of those. Take your time and thanks for creating this tool in the first place fellow WA-nian.

**Comment by vaygr** [comment]
Let me check this one and I'll get back to you.

**Comment by vaygr** [comment]
@firecat53 ready for review.

I didn't realize `get_database()` wasn't being executed in case no args are passed. In fact, we don't really need to associate TOTP mode with a database, but rather use a global flag here.

Works as expected now.

**Comment by firecat53** [comment]
So sorry for the delay again! One more issue and then I think we can get this merged.

1. Open any database, say 'passwords.kdbx'
2. Now try to open a second database from the CLI in TOTP-only mode. `keepmenu -t -d otps.kdbx`

It will show any OTPs from passwords.kdbx, not from otps.kdbx.

What does work:

1. Open passwords.kdbx
2. `Open another database` from the menu and select otps.kdbx
3. Now run `keepmenu -t` and it will show the otps from otps.kdbx.

Basically, it's not possible to switch the current view it totp-only mode from the command line.

Hopefully that makes sense!

**Comment by vaygr** [comment]
Thanks, I think I got it right now. Please check.

Just to confirm expected behavior: as soon as you pass another database via `-d`, the next time you run `keepmenu` *without* any args, it'll open that last database that was overriden by the `-d` flag the last time, not the original one, correct?

**Comment by firecat53** [comment]
Hmmm, I'm still seeing the same behavior as above:

1. Open passwords.kdbx (non-totp). `keepmenu`
2. Open totp db `keepmenu -t -d otps.kdbx`

It shows any otp entries in the passwords db, not the entries from otps.kdbx as you'd expect. 

I'm out of time for troubleshooting right now...back in a day or two.

**Comment by vaygr** [comment]
Sorry, forgot to add `__main__.py` to the commit. Should be up to date now after the last force push.

**Comment by firecat53** [comment]
Is your branch rebased on the latest main/develop? I can't tell if there are
some differences that are driving some weirdness.

1. Your branch doesn't have three of the entries in the test.kdbx database that
   have some various types of otp formats.
2. On my machine, your branch is failing one test using `make test` (python
   3.11 and also with 3.10.8 using pyenv).
3. When I open the test.kdbx database from the main/develop branch using your
   branch, it displays some of the extra attribute fields that should be hidden,
   and doesn't actually type anything for the TOTP entry (which shows either
   None or **** depending on the entry).

Hopefully I'm not doing something dumb! Tests pass on the main/develop branch
and I'm able to type the TOTP entries from the three entries that went missing
on your branch.

Thanks!

**Comment by vaygr** [comment]
Hey, no, since this was started more than 2 months ago I didn't rebase it until now. I noticed there were some TOTP changes regarding formats in between. I now rebased it against `main`, but I see a couple of tests failing on both my branch and `main`.

Could you try again and see if test.kdbx database is in good shape?

**Comment by vaygr** [comment]
Hey @firecat53, just double-checking if you think this is good to go.

**Comment by firecat53** [comment]
Merged! Sorry about the mess and the delay! Been a long several months :roll_eyes:  I'm apparently rusty with my dev skills. Thanks again for your work on this! Scott

**Comment by vaygr** [comment]
@firecat53 no worries at all! Thanks for looking into this and verifying. That iteration was valuable and discovered a few bugs in the original work.

I'll see what other small things I could work on.

**Comment by treeshateorcs** [comment]
how do i paste a totp secret into the field? it just says "Secret key?:" but doesn't accept pastes ☹️

**Comment by firecat53** [comment]
That's entirely dependent on the launcher you use. I use bemenu and the clipboard paste shortcut is `Ctrl-y`.
